### PR TITLE
Updating Gateway Intents Page

### PR DESCRIPTION
**Pull request by Kitosight**

### CODE REVIEW

**Review by priyanujgogoi-28 — CHANGES_REQUESTED** [review]
Hello NightNutSky,
If you can, please do update the GIF too and “How to enable Gateway Intents in BDFD App”. Previously, we have to manually enable the intents in the app. As of now, "Intents" are automatically synced!

File: src/guides/gatewayIntents.md [hunk 1]
```diff
@@ -3,28 +3,18 @@ This guide will explain to you how to use Gateway Intents in BDFD, and what they
...
 ## Use of Gateway Intents
 ### Presence Intent
 Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage) and [`$getUserStatus[]`](../bdscript/getUserStatus.md) to work.
 ### Server Members Intent
 Required for some callbacks to work (e.g. for [`$onJoined[]`](../callbacks/onJoined.md))
 ### Message Content Intent
-Required for text commands to work if your bot is verified. Without this intent, your bot can only be based on slash commands.
+Required for the bot to receive information about the message (such as embed data).\
+It's also required for text commands to work, otherwise your bot can be based only on slash commands. 
+
```
Comment (priyanujgogoi-28) [inline src/guides/gatewayIntents.md:83]:
```suggestion
```

**Review by priyanujgogoi-28 — CHANGES_REQUESTED** [review]
An explanation of about Presence Intent & Members Intent would also be good if you could explain a bit. Additionally, include `$getCustomStatus[]` function in Presence Intent function list and also, add a new line before a heading. For example:
```
### Presence Intent
Required ...

### Server Members Intent
Required ...

### Message Content Intent
Required ...
```

File: src/guides/gatewayIntents.md [hunk 2]
```diff
@@ -3,28 +3,18 @@ This guide will explain to you how to use Gateway Intents in BDFD, and what they
...
-![BDFD](https://user-images.githubusercontent.com/70456337/183294156-4ecc60b0-7112-4bf3-b833-39b8c2004fc6.gif)
-
 ## Use of Gateway Intents
 ### Presence Intent
 Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage) and [`$getUserStatus[]`](../bdscript/getUserStatus.md) to work.
 ### Server Members Intent
 Required for some callbacks to work (e.g. for [`$onJoined[]`](../callbacks/onJoined.md))
 ### Message Content Intent
-Required for text commands to work if your bot is verified. Without this intent, your bot can only be based on slash commands.
+Required for the bot to receive information about the message (such as embed data).\
```
Comment (MineBartekSA) [inline src/guides/gatewayIntents.md]: You can still receive the basic message information through the `MESSAGE_CREATE` gateway event, but you need this intent to receive the content data stored in the message.
```suggestion
Required for the bot to receive information stored in the message (such as message content or embed data).\
```

**Review by MineBartekSA — CHANGES_REQUESTED** [review]
Thank you, and please address @Priyanuj-Gogoi suggestions.

File: src/guides/gatewayIntents.md [hunk 3]
```diff
@@ -3,28 +3,21 @@ This guide will explain to you how to use Gateway Intents in BDFD, and what they
...
 
-### BDFD App
-1. Open the app and select your bot.
-2. Go to the "Settings" tab.
-3. Scroll down and enable the Gateway Intents you are interested in.
-4. Save changes.
-
-![BDFD](https://user-images.githubusercontent.com/70456337/183294156-4ecc60b0-7112-4bf3-b833-39b8c2004fc6.gif)
-
 ## Use of Gateway Intents
```
Comment (priyanujgogoi-28) [inline src/guides/gatewayIntents.md]:
```suggestion
## Privileged Gateway Intents
```

File: src/guides/gatewayIntents.md [hunk 4]
```diff
@@ -3,28 +3,21 @@ This guide will explain to you how to use Gateway Intents in BDFD, and what they
 If you are looking for general information about Discord's Gateway Intents, read it [here](https://discord.com/developers/docs/topics/gateway#gateway-intents).
 
 ## How to enable Gateway Intents
```
Comment (priyanujgogoi-28) [inline src/guides/gatewayIntents.md]:
```suggestion
## Enabling Gateway Intents
```

File: src/guides/gatewayIntents.md [hunk 5]
```diff
@@ -3,28 +3,21 @@ This guide will explain to you how to use Gateway Intents in BDFD, and what they
...
-3. Scroll down and enable the Gateway Intents you are interested in.
-4. Save changes.
-
-![BDFD](https://user-images.githubusercontent.com/70456337/183294156-4ecc60b0-7112-4bf3-b833-39b8c2004fc6.gif)
-
 ## Use of Gateway Intents
 ### Presence Intent
-Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage) and [`$getUserStatus[]`](../bdscript/getUserStatus.md) to work.
+Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage), [`$getUserStatus[]`](../bdscript/getUserStatus.md), and [`$getCustomStatus[]`](../bdscript/getCustomStatus.md) to work.\
+Allows bot to get information about users in the guild (such as current status).
```
Comment (priyanujgogoi-28) [inline src/guides/gatewayIntents.md]:
```suggestion
```

File: src/guides/gatewayIntents.md [hunk 6]
```diff
@@ -3,28 +3,21 @@ This guide will explain to you how to use Gateway Intents in BDFD, and what they
...
-2. Go to the "Settings" tab.
-3. Scroll down and enable the Gateway Intents you are interested in.
-4. Save changes.
-
-![BDFD](https://user-images.githubusercontent.com/70456337/183294156-4ecc60b0-7112-4bf3-b833-39b8c2004fc6.gif)
-
 ## Use of Gateway Intents
 ### Presence Intent
-Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage) and [`$getUserStatus[]`](../bdscript/getUserStatus.md) to work.
+Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage), [`$getUserStatus[]`](../bdscript/getUserStatus.md), and [`$getCustomStatus[]`](../bdscript/getCustomStatus.md) to work.\
```
Comment (priyanujgogoi-28) [inline src/guides/gatewayIntents.md]:
```suggestion
Allows the bot to receive `PRESENCE_UPDATE` event. This intent is primarily used to retrieve current user's presence state. For example, Activities (i.e PLAYING, LISTENING), Presence (i.e Online, Idle) and Custom status.

Functions which requires this intent are as follows:
- [`$getCustomStatus[]`](../bdscript/getCustomStatus.md)
- [`$getUserStatus[]`](../bdscript/getUserStatus.md)
- [`$membersCount[]`](../bdscript/membersCount.md#second-usage)
```

File: src/guides/gatewayIntents.md [hunk 7]
```diff
@@ -3,28 +3,21 @@ This guide will explain to you how to use Gateway Intents in BDFD, and what they
...
+Allows bot to get information about users in the guild (such as current status).
+
 ### Server Members Intent
-Required for some callbacks to work (e.g. for [`$onJoined[]`](../callbacks/onJoined.md))
+Required for some callbacks to work (e.g. for [`$onJoined[]`](../callbacks/onJoined.md)).\
+Allows bot to get a list of users in the guild.
```
Comment (priyanujgogoi-28) [inline src/guides/gatewayIntents.md]:
```suggestion
```

File: src/guides/gatewayIntents.md [hunk 8]
```diff
@@ -3,28 +3,21 @@ This guide will explain to you how to use Gateway Intents in BDFD, and what they
...
-
 ## Use of Gateway Intents
 ### Presence Intent
-Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage) and [`$getUserStatus[]`](../bdscript/getUserStatus.md) to work.
+Required for the functions [`$membersCount[]`](../bdscript/membersCount.md#second-usage), [`$getUserStatus[]`](../bdscript/getUserStatus.md), and [`$getCustomStatus[]`](../bdscript/getCustomStatus.md) to work.\
+Allows bot to get information about users in the guild (such as current status).
+
 ### Server Members Intent
-Required for some callbacks to work (e.g. for [`$onJoined[]`](../callbacks/onJoined.md))
+Required for some callbacks to work (e.g. for [`$onJoined[]`](../callbacks/onJoined.md)).\
```
Comment (priyanujgogoi-28) [inline src/guides/gatewayIntents.md]:
```suggestion
Allows the bot to receive `GUILD_MEMBER_ADD`, `GUILD_MEMBER_UPDATE, `GUILD_MEMBER_REMOVE`, and `THREAD_MEMBERS_UPDATE` events. This intent is primarily required to fetch entire list of guild members and getting specific guild member info (like guild joining, leaving, profile update etc.).

Callbacks which requires this intent are as follows:
- [`$onJoined[]`](../callbacks/onJoined.md)
- [`$onLeave[]`](../callbacks/onLeave.md)
```

**Review by priyanujgogoi-28 — CHANGES_REQUESTED** [review]
Hopefully, you can accept these changes. I have checked the Presence and Member Intent yet so far.

Also, i feel like we should add a short definition of "Gateway Intents" too and some more info about message intent (like Affecting functions)

**Comment by priyanujgogoi-28** [comment]
I'll try to check this a bit later for now.

File: src/guides/gatewayIntents.md (hunk 4 above)
Comment (priyanujgogoi-28) [reply src/guides/gatewayIntents.md]: Shortening of Headers as stated in [CONTRIBUTING.md](https://github.com/NilPointer-Software/bdfd-wiki/blob/dev/CONTRIBUTING.md#shortening-headers) file

**Review by priyanujgogoi-28 — APPROVED** [review]
Thanks!

File: src/guides/gatewayIntents.md [hunk 9]
```diff
@@ -1,30 +1,84 @@
...
+Gateway Intents can be further divided into:
+- Standard Intents[^1]
+- [Privileged Intents](#privileged-gateway-intents)
+
+For example, a bot needs to send the `GUILD_MESSAGES` standard intent and the `MESSAGE_CONTENT` privileged intent to receive the `MESSAGE_CREATE` event with the messsage content.
+
+> 📚 Learn more about Gateway Intents [here](https://discord.com/developers/docs/topics/gateway#gateway-intents).
+
+### Privileged Gateway Intents
+Privileged Gateway Intents are special intents which needs to be manually enabled in your bot application settings.
```
Comment (noituri) [inline src/guides/gatewayIntents.md]:
```suggestion
Privileged Gateway Intents are special intents which need to be manually enabled in your bot application settings.
```

**Review by noituri — CHANGES_REQUESTED** [review]
Thanks for contributing :)

**Review by noituri — APPROVED** [review]

**Review by MineBartekSA — APPROVED** [review]

**Pull request merged**